docs/JAX_basic_text_classification.md

```diff
@@ -0,0 +1,455 @@
+```{code-cell} ipython3
+print("Train encoded text batch info:", type(train_batch["text"]), train_batch["text"].shape, train_batch["text"].dtype)
+print("Train labels batch info:", type(train_batch["label"]), train_batch["label"].shape, train_batch["label"].dtype)
+```
+
+```{code-cell} ipython3
+print("Batch data:", train_batch["text"][:2, :12], train_batch["label"][:2])
+```
+
+## Model for text classification
+
+We choose a simple 1D convnet to classify the text. The first layer of the model transforms input tokens into float features using an embedding layer (`nnx.Embed`), then they are encoded further with convolutions. Finally, we classify encoded features using a fully-connected layers.
+
+```{code-cell} ipython3
+from typing import Callable
+
+import jax
+import jax.numpy as jnp
+from flax import nnx
+
+
+class TextConvNet(nnx.Module):
+    def __init__(
+        self,
+        vocab_size: int,
+        num_classes: int = 2,
+        embed_dim: int = 256,
+        hidden_dim: int = 320,
+        dropout_rate: float = 0.5,
+        conv_ksize: int = 12,
+        activation_layer: Callable = nnx.relu,
+        rngs: nnx.Rngs = nnx.Rngs(0),
+    ):
+        self.activation_layer = activation_layer
+        self.token_embedding = nnx.Embed(
+            num_embeddings=vocab_size,
+            features=embed_dim,
+            rngs=rngs,
+        )
+        self.dropout = nnx.Dropout(dropout_rate, rngs=rngs)
+        self.conv1 = nnx.Conv(
+            in_features=embed_dim,
+            out_features=hidden_dim,
+            kernel_size=conv_ksize,
+            strides=conv_ksize // 2,
+            rngs=rngs,
+        )
+        self.lnorm1 = nnx.LayerNorm(hidden_dim, rngs=rngs)
+        self.conv2 = nnx.Conv(
+            in_features=hidden_dim,
+            out_features=hidden_dim,
+            kernel_size=conv_ksize,
+            strides=conv_ksize // 2,
+            rngs=rngs,
+        )
+        self.lnorm2 = nnx.LayerNorm(hidden_dim, rngs=rngs)
+
+        self.fc1 = nnx.Linear(hidden_dim, hidden_dim, rngs=rngs)
+        self.fc2 = nnx.Linear(hidden_dim, num_classes, rngs=rngs)
+
+    def __call__(self, x: jax.Array) -> jax.Array:
+        # x.shape: (N, max_length)
+        x = self.token_embedding(x)
+        x = self.dropout(x)   # x.shape: (N, max_length, embed_dim)
+
+        x = self.conv1(x)
+        x = self.lnorm1(x)
+        x = self.activation_layer(x)
+        x = self.conv2(x)
+        x = self.lnorm2(x)
+        x = self.activation_layer(x)  # x.shape: (N, K, hidden_dim)
+
+        x = nnx.max_pool(x, window_shape=(x.shape[1], ))  # x.shape: (N, 1, hidden_dim)
+        x = x.reshape((-1, x.shape[-1]))  # x.shape: (N, hidden_dim)
+
+        x = self.fc1(x)  # x.shape: (N, hidden_dim)
+        x = self.activation_layer(x)
+        x = self.dropout(x)
+        x = self.fc2(x)  # x.shape: (N, 2)
+
+        return x
+
+
+x = jnp.ones((4, max_length), dtype="int32")
+module = TextConvNet(vocab_size)
+y = module(x)
+print(y.shape)
```

Comment:
Why is this shape informative here?

Reply:
To ensure that we output expected shape: (N, num_classes), updated to
```
print("Prediction shape (N, num_classes): ", y.shape)
```